docs/architecture/adr-006-poap-v2.md

```diff
@@ -0,0 +1,75 @@
+# ADR 006: POAP v2 Contract
+
+## Changelog
+
+- Jul 11, 2023: Initial draft;
+
+## Status
+PROPOSED Not Implemented
+
+## Abstract
+This ADR defines the new architecture of a POAP v2 Smart Contract that will replace the current `cw721-poap`, `poap-manager`, and `poap` smart contracts.
+
+## Context
+The name POAP stands for Proof of Attendance Protocol, which is a simple NFT that attests to the fact that a user has participated in a particular event.
+
+Currently, we have implemented the ability to create and manage POAPs with three smart contracts: `cw721-poap`, `poap-manager`, and `poap`. Although this division works and makes sense under the Single Responsibility Principle, there are several downsides to it. First, users can easily upload and instantiate multiple instances of each of those contracts, leading to an unnecessary amount of contracts on the chain. Second, it makes it harder for newcomers to understand the interactions between the three contracts. Lastly, having multiple smart contracts for something as simple as a POAP seems excessive, as ideally, a POAP should have minimal operations.
+
+## Decision
+The decision is to merge all POAP-related operations into a single smart contract called `poap`, replacing all existing POAP-related smart contracts.
+
+## Specifications
+Below are the specifications for the various messages that the `poap` contract should support.
+
+### Messages
+
+#### Instantiate
+To instantiate this smart contract, the user needs to provide:
+
+* `admin` address: Identifies the user that controls the contract.
```

Comment:
Is the admin also able to mint POAP to other users?
What do you think about the possibility to add a `minter` account?

Reply:
I have now added the ability to mint tokens for other users. 

However, what would be the utility of adding a `minter` account?

Reply:
Maybe it could be easier to explain the need with an example:

By thinking of the ETH POAP version, the admin of POAP is a company (for instance, CorporateA). They create the master POAP (which involves instantiating the contract, essentially calling the instantiate function) from their wallet. They use their own account with their own funds.

Once they've created it, they ask POAP Company for the ability to mint the POAPs for their users. In this scenario, of course, POAP Company does not know the mnemonic of CorporateA and it is not authorized to modify the master POAP, while it is authorized to mint the tokens.

Does it make sense to you?

Reply:
Yes, thanks. It acuallty provides a usecase I didn't think about. I have updated the ADR to also consider this.